providers/src/airflow/providers/openlineage/utils/operator.py

```diff
@@ -0,0 +1,213 @@
+                log.error("Key %s not in received event %s\nExpected %s", k, result, expected)
+                return False
+            if not match(v, result[k], env):
+                log.error(
+                    "For key %s, expected value %s not equals received %s\nExpected: %s, request: %s",
+                    k,
+                    v,
+                    result[k],
+                    expected,
+                    result,
+                )
+                return False
+    elif isinstance(expected, list):
+        if len(expected) != len(result):
+            log.error("Length does not match: expected %d, result: %d", len(expected), len(result))
+            return False
+        for i, x in enumerate(expected):
+            if not match(x, result[i], env):
+                log.error(
+                    "List not matched at %d\nexpected:\n%s\nresult: \n%s",
+                    i,
+                    json.dumps(x),
+                    json.dumps(result[i]),
+                )
+                return False
+    elif isinstance(expected, str):
+        if "{{" in expected:
+            # Evaluate jinja: in some cases, we want to check only if key exists, or if
+            # value has the right type
+            try:
+                rendered = env.from_string(expected).render(result=result)
+            except ValueError as e:
+                log.error("Error rendering jinja template %s: %s", expected, e)
+                return False
+            if rendered == "true" or rendered == result:
+                return True
+            log.error("Rendered value %s does not equal 'true' or %s", rendered, result)
+            return False
+        elif expected != result:
+            log.error("Expected value %s does not equal result %s", expected, result)
+            return False
+    elif expected != result:
+        log.error("Object of type %s: %s does not match %s", type(expected), expected, result)
+        return False
+    return True
+
+
+class OpenLineageTestOperator(BaseOperator):
+    """
+    This operator is added for system testing purposes.
+
+    It compares expected event templates set on initialization with ones emitted by OpenLineage integration
+    and stored in Variables by VariableTransport.
+    :param event_templates: dictionary where key is the key used by VariableTransport in format of <DAG_ID>.<TASK_ID>.event.<EVENT_TYPE>, and value is event template (fragment) that need to be in received events.
+    :param file_path: alternatively, file_path pointing to file with event templates will be used
+    :param env: jinja environment used to render event templates
+    :raises: ValueError if the received events do not match with expected ones.
+    """
+
+    def __init__(
+        self,
+        event_templates: dict[str, dict] | None = None,
+        file_path: str | None = None,
+        env: Environment = setup_jinja(),
+        multiple_events: bool = False,
+        delete: bool = True,
```

Comment:
```suggestion
        clear_variables: bool = True,
```
Maybe some more descriptive name can be used here?

Reply:
Fixed, thanks.